src/Ractive/prototype/set.js

```diff
@@ -10,9 +10,9 @@ export default function Ractive$set ( keypath, value ) {
 	if ( isObject( keypath ) ) {
 		const map = keypath;
 
-		for ( keypath in map ) {
-			if ( map.hasOwnProperty( keypath) ) {
-				set( this, keypath, map[ keypath ] );
+		for ( const k in map ) {
+			if ( map.hasOwnProperty( k) ) {
+				set( this, k, map[k] );
 			}
```

Comment:
> const keypaths = Object.keys( map );
> let i = keypaths.length;
> while ( i-- ) {
>   set( this, keypaths[i], map[ keypaths[i] ] );
> }

yes, that should be the fastest :) - consistent with all the testing I did around keys and iterations